tests/Doctrine/ODM/MongoDB/Tests/Functional/Ticket/GH2339Test/ParentDocument.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ODM\MongoDB\Tests\Functional\Ticket\GH2339Test;
+
+use Doctrine\ODM\MongoDB\Mapping\Annotations as ODM;
+use MongoDB\BSON\ObjectIdInterface;
+
+/**
+ * @ODM\Document
+ */
+class ParentDocument
+{
+    /** @ODM\Id */
+    protected ObjectIdInterface $id;
+
+    /**
+     * @ODM\EmbedMany(targetDocument=InlinedDocument::class)
+     *
+     * @var InlinedDocument[]
+     */
+    protected array $inlined = [];
+
+    public function getId(): ObjectIdInterface
+    {
+        return $this->id;
+    }
+
+    public function addInlined(InlinedDocument $document)
```

Comment:
We require at least PHP 7.2 so we can safely add `void` type hint.
```suggestion
    public function addInlined(InlinedDocument $document): void
```